pkg/ekcoops/operator.go

```diff
@@ -127,19 +127,32 @@ func (o *Operator) reconcileRook(nodes []corev1.Node, doFullReconcile bool) erro
 		readyCount, err := o.ensureAllUsedForStorage(nodes)
 		if err != nil {
 			multiErr = multierror.Append(multiErr, errors.Wrapf(err, "ensure all ready nodes used for storage"))
-		} else {
+		}
+
+		if err != nil {
 			err := o.adjustPoolReplicationLevels(readyCount, doFullReconcile)
 			if err != nil {
 				multiErr = multierror.Append(multiErr, errors.Wrapf(err, "adjust pool replication levels"))
 			} else {
-				err := o.controller.ReconcileMonCount(readyCount)
+				err := o.controller.ReconcileMonCount(context.TODO(), readyCount)
 				if err != nil {
 					multiErr = multierror.Append(multiErr, errors.Wrapf(err, "reconcile mon count"))
 				}
+				err = o.controller.ReconcileMgrCount(context.TODO(), readyCount)
+				if err != nil {
+					multiErr = multierror.Append(multiErr, errors.Wrapf(err, "reconcile mgr count"))
+				}
 			}
 		}
```

Comment:
Why not just put this code in the first `if err !=nil` block?

Reply:
Oops. It should have been `if err == nil {`. Fixed